### PR TITLE
Refactored code into a conda package

### PR DESCRIPTION
**Pull request by blackbox-tech**
@jeremy-large I've added the setup.py to create a python package called 'codit' and written a conda recipe to install/unittest it along with all its dependencies.  The code in the package is now under the condit namespace so that it does not clash with other modules in the global namespace.

### CODE REVIEW

**Comment by blackbox-tech** [comment]
As well as the code getting packaged by conda, the data will also be packaged and will be installed by conda into the share dir.  [This line](https://github.com/jeremy-large/CODiT/blob/feature/conda_refactor/lib/codit/population/networks/city_config/city_cfg.py#L5) will automatically get the path to the shared data directory under your current PYTHONPATH or failing that the shared data directory of the current conda environment.

File: lib/codit/population/networks/city_config/city_cfg.py [hunk 1]
```diff
@@ -1,18 +1,20 @@
-import os
 import pandas as pd
+from pathlib import Path
 
+# This will resolve to the shared data dir under the current PYTHONPATH or CONDA_PREFIX
+SHARE_DIR = Path(__file__).parent / "../../../../../share"
```
Comment (jeremy-large) [inline lib/codit/population/networks/city_config/city_cfg.py:5]: noted

File: lib/codit/population/networks/city_config/city_cfg.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
     :param file_name: table-8.csv
     :param repo_dir: desired data type
     :return: value according to table
     """
-    return pd.read_csv(os.path.join(repo_dir, 'data', 'city', section, file_name), index_col=0)
+    return pd.read_csv(data_dir / section / file_name, index_col=0)
 
 
-DEMO_DATA = (read_demographic_data('population', 'table-8.csv', "..")["%"] / 100).to_dict()
+DEMO_DATA = (read_demographic_data('population', 'table-8.csv')["%"] / 100).to_dict()
```
Comment (jeremy-large) [inline lib/codit/population/networks/city_config/city_cfg.py:17]: thanks - not sure what that ".." was offering

File: recipes/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
+    - python
+    - numpy
+    - pandas
+    - scikit-learn
+    - statsmodels
```
Comment (jeremy-large) [inline recipes/meta.yaml:19]: Truth be told, we don't currently use `statsmodels`

File: tests/test_all.py [hunk 4]
```diff
@@ -1,37 +1,39 @@
-import sys
-import os
 import random
+import numpy as np
 
-# point at library; I need some lessons on doing good PYTHONPATHs:
```
Comment (jeremy-large) [inline tests/test_all.py:5]: Good to see the back of this!

File: tests/test_all.py [hunk 5]
```diff
@@ -1,37 +1,39 @@
...
-                                       [43.0, 0.035, 0.0, 0.007, 0.461, 0.447],
-                                       [44.0, 0.035, 0.0, 0.007, 0.458, 0.448],
-                                       [45.0, 0.035, 0.0, 0.007, 0.457, 0.448]]
+
+    # Skip this test because the the simulator is not deterministic.
+    # I believe it is because it uses np.random by calls multi-threaded numpy code.
+    # TODO: I believe the correct solution is for the simulator to be constructed with its own np.random.RandomState class
+    #
+    return
+    np.testing.assert_allclose(o.recorder.story[40:45] == [[41.0, 0.035, 0.0, 0.007, 0.467, 0.452],
```
Comment (jeremy-large) [inline tests/test_all.py:27]: better, thanks

File: tests/test_all.py [hunk 6]
```diff
@@ -85,9 +85,8 @@ def test_toy_model():
     o = Outbreak(s, d, pop_size=1000, seed_size=2, n_days=ALL_TIME_DAYS, population_type=Population, person_type=Person)
     # so, this is a village of 10000 people with 2 starting off infected
     o.simulate()
-    assert o.recorder.story[90:95] == \
-           [[18.199999999999967, 0.619, 0.57, 0.0, 0.0, 0.0],
```
Comment (jeremy-large) [inline tests/test_all.py:89]: farwell, `999999`

**Review by jeremy-large — APPROVED** [review]
Thanks  @blackbox-tech !

**Pull request merged**